### ISSUE TITLE
Rule request: empty-constructor

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#3631]
#### TypeScript code being linted

```typescript
class Counter {
    private count: number = 0;

    public constructor() { }

    public increase(): number {
        return this.count += 1;
    }
}
```

#### Actual behavior

No complaints under `tslint:recommended`.

#### Expected behavior

The constructor does nothing and should be removed. There's no complaint saying this.

Folks new to classes / TypeScript will sometimes add constructors because they think they're necessary no matter what.

**Comment by JoshuaKGoldberg** [comment]
Another thing to look out for as an option (should just initialize via `private count: number = 0;`):

class Counter {
    private count: number;

    public constructor() {
        this.count = 0;
    }

    public increase(): number {
        return this.count += 1;
    }
}
```

**Comment by vpanjganj** [comment]
@JoshuaKGoldberg isn't this covered by `[tslint] block is empty [no-empty]` rule anyway?

**Comment by JoshuaKGoldberg** [comment]
@vpanjganj technically yes, but not everybody enables `no-empty`.

**Comment by asmironov** [comment]
And what about the following example? Isn't it unnecessary as well?
```ts
constructor() {
    super();
}
```

**Comment by JoshuaKGoldberg** [comment]
@asmironov yup, that's correct!

**Comment by Bublade** [comment]
@asmironov @JoshuaKGoldberg isnt something like this unnecessary as well?
```Typescript
constructor(requiredParam: string) {
    super(requiredParam);
}
```

**Comment by JoshuaKGoldberg** [comment]
Ah, this rule was added in #3647. It only handles the case of a constructor with an empty body.

@Bublade that's correct! If you'd like to file a new issue, that'd be great.

**Comment by elharony** [comment]
This seems like a +6 yrs concern since #67
As mentioned in the old issue, to get around this, I am adding an empty comment
```ts
constructor() {
  //
}

```

**Comment by elharony** [comment]
@JoshuaKGoldberg It seems like a +6 yrs concern since #67 🤔 
As mentioned in the old issue, to get around this, you can add an empty comment
```ts
constructor() {
  //
}
```

For new devs who write in TypeScript and came across this comment, just listen to @JoshuaKGoldberg 's tip, and delete "empty" constructor, if it's empty, then you don't need it, and removing redundant stuff is a great benefit we get by using a linter. **Do NOT try to skip/ignore/stop every linting rule to "pass" it!**
